crates/number-conversions/src/lib.rs

```diff
@@ -56,6 +66,12 @@ pub fn big_decimal_to_u256(big_decimal: &BigDecimal) -> Option<U256> {
     big_int_to_u256(&big_int).ok()
 }
 
+pub fn rational_to_big_decimal<T: Clone + Integer + ToPrimitive + ToBigInt>(
+    value: &Ratio<T>,
+) -> Option<BigDecimal> {
+    BigDecimal::from_f64(value.to_f64()?)
```

Comment:
I _think_ this might lead to precision issues for Balancer (as the denominator is 1e18).

Maybe we should do:

```suggestion
    BigDecimal::new(value.numer(), 0) / BigDecimal::new(value.denom(), 0)
```

Reply:
How *does* BigDecimal represent something like `1/3` by the way?

Reply:
Fixed in [4706a97](https://github.com/cowprotocol/services/pull/1251/commits/4706a97f56b970a144062acd6bda0486cf96a950).

> How does BigDecimal represent something like 1/3 by the way?

It would print repeating numbers with a lot of precision:
```
0.3333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333
```
So far I only filled in the blanks of the API that was already defined. I'm not sure if this is a conscious trade-off or if it was an oversight when defining the interface. Do you think this will be fine or should we move to `Ratio<T>` for everything that is not an integer?

Reply:
Concentrated liquidity fields are always in BPS, so that means a `num::Ratio<u32>` where `assert_eq!(*num.denom(), 10000)`. Because of this, a `BigDecimal` is a perfectly appropriate representation for concentrated liquidity fees (`f64` not, e.g. `0.3_f64 != 0.3`).

The reason for choice of `BigDecimal` in the JSON was purely esthetic: reading `"0.005"` is more natural and more easily associated with the "50 BPS" fee than `[1, 200]` (noting that `num::Ratio` serializes as a tuple and always reduces its numerator and denominator).

In general I agree that `BigDecimal` is not OK for fractional fees, but here we really do have "decimal fees".